src/interpreter_funcs.rs

```diff
@@ -415,12 +415,25 @@ impl Func for FuncImplSynchronizeMeshFaces {
             && mesh_analysis::is_mesh_manifold(&edge_sharing_map)
         {
             let unoriented_edges: Vec<_> = geometry.unoriented_edges_iter().collect();
```

Comment:
You're right, they should be deduplicated. Nevertheless, nothing fails if they are not, just the calculations might be less efficient. 
I prefer some sort of deduplication already in the `unoriented_edges_iter` method. Once we get rid of the custom `Eq` on the `UnorientedEdge`, this will be also more efficient.
When changing the `UnorientedEdge` I'd like to add a getter for `OrientedEdge`s, `UnorientedEdge`s and some `oriented_edge_index_to_unoriented_edge_index_map`. They are and will be useful together and they are also quite efficient to be calculated at the same time.